### DISCUSSION TITLE
Jamulus 3.9.0beta Windows latency issues

### DISCUSSION THREAD

**Discussion by bevbones** [jamulussoftware/jamulus#2717] [Bug Investigation]
I am a regular user of jamulus, running a server in 3.8.2 for the last few months with minimal latency and no issues. I tried the new beta release and found when i connected to the server on the same machine as i usually would, i experienced a significant increase in latency. Other users didn't seem to experience this issue, just me connecting on the same machine. I used the same network, and programme settings as i had previously. Everything running at 64 samples. After reverting to the 3.8.2 version I experienced no latency issues. 

Windows 11 OS if thats any use. 

If it's user error please let me know!

**Comment by dcorson-ticino-com** [comment]
Increased latency running both the server and the client on the same machine under Windows we have seen before (#1855).
I am not aware of any solution.
The work around is to not run the client and server on the same machine.

**Comment by pcar75** [comment]
Also have a look at some w11 observations in "With Jamulus, plan to add or move-to Linux Ubuntu ? #2518" and elsewhere on internet. It may be a version to avoid ...

**Comment by pljones** [comment]
@bevbones, could you say how much "significant" was - do you mean up from 0-2ms to 10-20ms?  Or 0-2ms to 2-4ms?  Or something else?

Does this happen when you connect to other 3.9.0beta1 Servers?  (To test, you can use Server Address  `updatecheck2.jamulus.io` as a 3.8.2 version and Server Address `rock.jamulus.io:22324` (which is "Directory on 22324" in the Genre Rock) as a 3.9.0beta1 version.  These run on the same machine, so you should get the same ping and latency reported, more or less.)

Thanks.

**Reply by pljones** [reply]
I've just tried the following:
* Windows machine (3.9.0beta1) to Linux machine (3.8.2 and 3.9.0beta1) - same LAN
2ms difference - the 3.8.2 connection being better
* Linux machine (3.8.2 and 3.9.0beta1) to Linux machine (3.8.2 and 3.9.0beta1) - same machine
No more than 2ms difference - with the 3.9.0beta1 to 3.9.0beta1 having the lowest latency (10ms - same as the best I got for 3.8.2)

Given the low difference (2ms) and I see that between sessions on my Windows machines all the time, I have to say I can't reproduce the reported problem.

(Ping in all the above cases was 0-1ms.)

**Reply by bevbones** [reply]
Thanks for all the replies. I'll test again and see if I can reproduce the latency.  It felt about 10ms. And I'll try running the server on a different machine to see if i can improve the performance of the clients/server. All the above is super helpful
